#GATK Haplotype Caller -- GATK v3.5
#Adjustments to options needed for each run:
#set chromosome, -L
#input bam, -isorted_bam
#input bai, -isorted_bai
#output gvcf filename prefix, -ioutput_prefix
#output path, --destination

#HG002 run 5/17/16, ran ~30min - 5hrs
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.1.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.1.bai -ioutput_prefix=HG002_1_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.2.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.2.bai -ioutput_prefix=HG002_2_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.3.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.3.bai -ioutput_prefix=HG002_3_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.4.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.4.bai -ioutput_prefix=HG002_4_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.5.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.5.bai -ioutput_prefix=HG002_5_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.6.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.6.bai -ioutput_prefix=HG002_6_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.7.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.7.bai -ioutput_prefix=HG002_7_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.8.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.8.bai -ioutput_prefix=HG002_8_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.9.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.9.bai -ioutput_prefix=HG002_9_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.10.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.10.bai -ioutput_prefix=HG002_10_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.11.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.11.bai -ioutput_prefix=HG002_11_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.12.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.12.bai -ioutput_prefix=HG002_12_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.13.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.13.bai -ioutput_prefix=HG002_13_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.14.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.14.bai -ioutput_prefix=HG002_14_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.15.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.15.bai -ioutput_prefix=HG002_15_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.16.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.16.bai -ioutput_prefix=HG002_16_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.17.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.17.bai -ioutput_prefix=HG002_17_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.18.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.18.bai -ioutput_prefix=HG002_18_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.19.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.19.bai -ioutput_prefix=HG002_19_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.20.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.20.bai -ioutput_prefix=HG002_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.21.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.21.bai -ioutput_prefix=HG002_21_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.22.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.22.bai -ioutput_prefix=HG002_22_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.X.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.X.bai -ioutput_prefix=HG002_X_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L X -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.Y.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.Y.bai -ioutput_prefix=HG002_Y_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L Y -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.MT.bam -isorted_bai=GIAB:/HG002/Illumina/HG002.hs37d5.300x_novoalign.MT.bai -ioutput_prefix=HG002_MT_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L MT -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG002/Illumina/GATKHC_output/

