#HG001 Illumina 150bp300X mapped with GRCh38
#GATK Haplotype caller
#Run 9/22/16 Ran 1:30 to 6 hr 50 min
#chr X run on 10/5
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_1.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_1.bai -ioutput_prefix=HG001_1_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_2.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_2.bai -ioutput_prefix=HG001_2_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_3.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_3.bai -ioutput_prefix=HG001_3_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_4.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_4.bai -ioutput_prefix=HG001_4_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_5.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_5.bai -ioutput_prefix=HG001_5_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_6.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_6.bai -ioutput_prefix=HG001_6_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_7.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_7.bai -ioutput_prefix=HG001_7_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_8.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_8.bai -ioutput_prefix=HG001_8_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_9.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_9.bai -ioutput_prefix=HG001_9_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_10.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_10.bai -ioutput_prefix=HG001_10_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_11.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_11.bai -ioutput_prefix=HG001_11_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_12.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_12.bai -ioutput_prefix=HG001_12_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_13.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_13.bai -ioutput_prefix=HG001_13_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_14.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_14.bai -ioutput_prefix=HG001_14_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_15.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_15.bai -ioutput_prefix=HG001_15_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_16.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_16.bai -ioutput_prefix=HG001_16_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_17.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_17.bai -ioutput_prefix=HG001_17_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_18.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_18.bai -ioutput_prefix=HG001_18_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_19.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_19.bai -ioutput_prefix=HG001_19_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_20.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_20.bai -ioutput_prefix=HG001_20_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_21.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_21.bai -ioutput_prefix=HG001_21_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_22.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_22.bai -ioutput_prefix=HG001_22_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_X.bam -isorted_bai=/HG001/GRCh38/Illumina/HG001_GRCh38_Ilmn_150bp300x_novoalign_X.bai -ioutput_prefix=HG001_X_GRCh38_novoalign_150bp300x_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chrX -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/Illumina/GATKHC_output/
