
################################### I L L U M I N A ######################################

#Split novoalign bam file by chrom and add read groups for GATK
#bam files generated by NHGRI and available at: ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/NA12878/NIST_NA12878_HG001_HiSeq_300x/NHGRI_Illumina300X_novoalign_bams/HG001.hs37d5.300x.bam and ....bai
#Note that we do this in 3 separate processes on DNAnexus to parallelize, but this isn't required as long as there is sufficient disk space (~1.2TB)
#Sorted mappings (sorted_bam): HG001.hs37d5.300x_novoalign.bam
#Sorted mappings index (index_bai): HG001.hs37d5.300x_novoalign.bai
#Read group ID (rgid): 1
#Read group LB (rglb): all
#Read group PL (rgpl): illumina
#Read group PU (rgpu): all
#Read group SM (rgsm): NA12878
#Output examples for first 3 chrom's:
#Index1 (bai1): HG001.hs37d5.300x_novoalign.1.bai
#Index2 (bai2): HG001.hs37d5.300x_novoalign.2.bai
#Index3 (bai3): HG001.hs37d5.300x_novoalign.3.bai
#Bam1 (bam1): HG001.hs37d5.300x_novoalign.1.bam
#Bam2 (bam2): HG001.hs37d5.300x_novoalign.2.bam
#Bam3 (bam3): HG001.hs37d5.300x_novoalign.3.bam
dx run GIAB:/Workflow/samtools_splitchrom_addrg_1to5 -isorted_bam=HG001.hs37d5.300x_novoalign.bam -iindex_bai=HG001.hs37d5.300x_novoalign.bai -irgid=1 -irglb=all -irgpl=illumina -irgpu=all -irgsm=NA12878 
dx run GIAB:/Workflow/samtools_splitchrom_addrg_6to12 -isorted_bam=HG001.hs37d5.300x_novoalign.bam -iindex_bai=HG001.hs37d5.300x_novoalign.bai -irgid=1 -irglb=all -irgpl=illumina -irgpu=all -irgsm=NA12878 
dx run GIAB:/Workflow/samtools_splitchrom_addrg_13toMT -isorted_bam=HG001.hs37d5.300x_novoalign.bam -iindex_bai=HG001.hs37d5.300x_novoalign.bai -irgid=1 -irglb=all -irgpl=illumina -irgpu=all -irgsm=NA12878 

#GATK Haplotype Caller
#Adjustments to options needed for each run:
#set chromosome, -L
#input bam, -isorted_bam
#input bai, -isorted_bai
#output gvcf filename prefix, -ioutput_prefix
dx run GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=HG001.hs37d5.300x_novoalign.20.bam -isorted_bai=HG001.hs37d5.300x_novoalign.20.bai -ioutput_prefix=HG001_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf -iextra_options="-L 20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"

#GATK gvcf --> vcf conversion
#Adjustments to options needed for each run:
#input "vcf", -ivcfs
#input .vcf.tbi, -ivcfs
#output vcf filename prefix (adjust for RL and chemistry), -iprefix
dx run GIAB:/Workflow/GATK_V3.5/gatk-genotype-gvcfs-v3.5 -ivcfs=HG001_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf.vcf.gz -ivcfs=HG001_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC_gvcf.vcf.gz.tbi -iprefix=HG001_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC

#FreeBayes
#Adjustments to options needed for each run:
#input bam, -isorted_bam
#output vcf filename prefix, -ioutput_prefix
#set reference genome, -igenome_fastagz
#set chromosome for bamtools, -itargets_bed
dx run freebayes -isorted_bams=HG001.hs37d5.300x_novoalign.20.bam -ioutput_prefix=HG001_20_hs37d5_novoalign_Ilmn150bp300X_FB_regiontest -igenome_fastagz="Reference Genome Files:/H. Sapiens - GRCh37 - hs37d5 (1000 Genomes Phase II)/hs37d5.fa.gz" -igenotype_qualities=TRUE -itargets_bed="GIAB:/Workflow/Chromosome_bed_files/20.bed" -istandard_filters=FALSE -iadvanced_options="-F 0.05 -m 0"

#GATK CallableLoci
#Adjustments to options needed for each run:
#input bam, -iinput_bam
#input bai, -iinput_bai
#output files prefix, -ioutput_prefix
#set region/chromosome, -L 
#maximum coverage threshold (dependent on platform-- 2x median depth=560 for Illumina), -maxDepth
dx run GIAB:/Workflow/GATK_V3.5/gatk-callableloci-v3.5 -iinput_bam=HG001.hs37d5.300x_novoalign.20.bam -iinput_bai=HG001.hs37d5.300x_novoalign.20.bai -ioutput_prefix=HG001_hs37d5_novoalign_Ilmn150bp300X_callableloci -iextra_options="-L 20 -minDepth 20 -mmq 20 -maxDepth 560"


###################################### S O L I D ######################################### 

#Need to add preprocess of BAMS -- Combining runs for given genomes, adding read groups and splitting by chromosome

#GATK Haplotype Caller
#Adjustments to options needed for each run:
#set chromosome, -L
#input bam, -isorted_bam
#input bai, -isorted_bai
#output gvcf filename prefix (adjust for RL and chemistry), -ioutput_prefix
dx run GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/NA12878/SOLID/SOLID_20_RG.sorted.deduplicated.reorder.realigned.bam -isorted_bai=SOLID_20_RG.sorted.deduplicated.reorder.realigned.bam.bai -ioutput_prefix=HG001_20_h37_SOLiD_GATKHC_gvcf -iextra_options="-L 20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"

#GATK gvcf --> vcf conversion
#Adjustments to options needed for each run:
#input "vcf", -ivcfs
#input .vcf.tbi, -ivcfs
#output vcf filename prefix, -iprefix
dx run GIAB:/Workflow/GATK_V3.5/gatk-genotype-gvcfs-v3.5 -ivcfs=HG001_20_h37_SOLiD_GATKHC_gvcf.vcf.gz -ivcfs=HG001_20_h37_SOLiD_GATKHC_gvcf.vcf.gz.tbi -iprefix=HG001_20_h37_SOLiD_GATKHC

#GATK CallableLoci
#Adjustments to options needed for each run:
#input bam, -iinput_bam
#input bai, -iinput_bai
#output files prefix, -ioutput_prefix
#set region/chromosome, -L 
#maximum coverage threshold (dependent on platform-- 2x median (12) = 24 for SOLiD), -maxDepth
dx run GIAB:/Workflow/GATK_V3.5/gatk-callableloci-v3.5 -iinput_bam=IAB:/NA12878/SOLID/SOLID_20_RG.sorted.deduplicated.reorder.realigned.bam -iinput_bai=SOLID_20_RG.sorted.deduplicated.reorder.realigned.bam.bai -ioutput_prefix=HG001_20_h37_SOLiD_callableloci -iextra_options="-L 20 -minDepth 20 -mmq 20 -maxDepth 24"


######################### C O M P L E T E  G E N O M I C S ###############################    

#Generate vcf and bed from Complete Genomics vcfBeta
#vcfbeta file for NA12878 downloaded from ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/NA12878/analysis/CompleteGenomics_RMDNA_11272014/ASM/vcfBeta-GS000025639-ASM.vcf.bz2
#Adjustments to options needed for each run:
#split by chromosome, -ichrom
#location for output files, -destination
dx run GIAB:/Workflow/integration-prepare-cg -ivcf_in=/NA12878/Complete_Genomics/vcfBeta-GS000025639-ASM.vcf.bz2 -ichrom=20 --destination=/NA12878/Complete_Genomics/

  
################################## I O N  T O R R E N T ##################################  

#GATK 3.5 callableLoci
#Adjustments to options needed for each run:
#Split bed by chromosome, -L 
#adjust ouput name for each chromosome, -output_prefix
dx run GIAB:/Workflow/GATK_V3.5/gatk-callableloci-v3.5 -iinput_bam=GIAB:/NA12878/Ion_Torrent/IonXpress_020_rawlib.b37.bam -iinput_bai=GIAB:/NA12878/Ion_Torrent/IonXpress_020_rawlib.b37.bam.bai -ioutput_prefix=HG001_20_hs37d5_IonExome_callableloci -iextra_options="-L 20 -minDepth 20 -mmq 20"

#Generate formatted vcf and bed from Ion vcf and targets.bed and callableloci.bed
#NA12878 Ion vcf at ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/NA12878/analysis/IonTorrent_TVC_06302015/TSVC_variants_defaultlowsetting.vcf
#Ion exome targets bed at ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/NA12878/analysis/IonTorrent_TVC_06302015/AmpliseqExome.20141120_effective_regions.bed
#Adjustments to options needed for each run:
#bed file for desired chromosome, -icallablelocibed
#split by chromosome, -ichrom
#location for output files, -destination
dx run GIAB:/Workflow/integration-prepare-ion -ivcf_in=/NA12878/Ion_Torrent/TSVC_variants_defaultlowsetting.vcf -icallablelocibed=/NA12878/Ion_Torrent/callableLoci_output/HG001_20_hs37d5_IonExome_callableloci.bed -itargetsbed=/NA12878/Ion_Torrent/AmpliseqExome.20141120_effective_regions.bed -ichrom=20 --destination=/NA12878/Ion_Torrent/

############################ I N T E G R A T I O N  v 3 . 1 ##############################

#Form high-confidence vcf and bed by integrating all individual vcf and bed files (excluding SOLID)
#multiple input vcfs from different datasets and variant callers: -ivcfs
#multiple input bed from different datasets, either output of GATK callableloci or not callable regions: -ibeds
#multiple annotations files from variant callers describing annotations for one-class filtering model: -iannotations
#tab-delimited file with vcf, bed, and annotations file names and which technology, dataset, and callset: -icallsettable
#putative structural variant regions to exclude from high-confidence regions: -isvbed
#split by chromosome, -ichrom
#location for output files, -destination

dx run GIAB:/Workflow/nist-integration-v3.1 -ivcfs=/NA12878/Complete_Genomics/Integration_prepare_cg_output/vcfBeta-GS000025639-ASM_20.vcf.gz -ivcfs=/NA12878/Illumina/GATKHC_output/HG001_20_hs37d5_novoalign_Ilmn150bp300X_GATKHC.vcf.gz -ivcfs=/NA12878/Illumina/FreeBayes_output/HG001_20_hs37d5_novoalign_Ilmn150bp300X_FB.vcf.gz -ivcfs=/NA12878/Ion_Torrent/Integration_prepare_ion_output/TSVC_variants_defaultlowsetting_20.vcf.gz -ibeds=/NA12878/Complete_Genomics/Integration_prepare_cg_output/vcfBeta-GS000025639-ASM_notcallable_20.bed -ibeds=/NA12878/Illumina/CallableLoci_output/HG001_20_hs37d5_novoalign_Ilmn150bp300X_callableloci.bed -ibeds=/NA12878/Ion_Torrent/Integration_prepare_ion_output/TSVC_variants_defaultlowsetting_notcallable_20.bed -iannotations=/NA12878/GATK_Annotations.txt -iannotations=/NA12878/Freebayes_Annotations.txt -iannotations=/NA12878/CG_Annotations.txt -iannotations=/NA12878/Ion_Annotations.txt -icallsettable=/NA12878/NA12878_RM8398_Datasets_CG_Illumina_Ion_Files_20.txt -isvbed=/NA12878/dbvar_NA12878_CNVs_merged_all_sort.bed -ichrom=20 -iprefix=NA12878_GIAB_highconf_CG-IllFB-IllGATKHC-Ion_20_v3.1b --destination=/NA12878/

   