#!/bin/bash

zgrep 'trf\|Simple_repeat\|Low_complex' /Users/jzook/Downloads/repeats.bed.gz | sed 's/^chr//' | cut -f1-3 | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 > /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50.bed
gunzip -c /Users/jzook/Downloads/segdups_selfchain.bed.gz | sed 's/^chr//' | cut -f1-3 | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 > /Users/jzook/Downloads/segdups_selfchain_merged50.bed

##Create bed file with regions covered by one contig or a variant called from each haploid assembly
for v in *.l100c500.no_ns.vcf.gz; do
  prefix=`echo $v | sed 's/.l100c500.no_ns.vcf.gz//'`
  zgrep -v ^# "$prefix".l100c500.no_ns.vcf.gz | sed 's/.*REFWIDENED=\(.*\):\(.*\)-\(.*\);ALTPOS.*/\1	\2	\3/' | awk '{FS=OFS="\t"} { print $1, $2-100, $3+100}' | grep -v '^GL\|^NC\|^hs\|^MT' | sed 's/^X/23/;s/^Y/24/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^24/Y/' > "$prefix".l100c500.no_ns.refwidenedplus100.bed
  gunzip -c "$prefix"_l100c500.qdelta.sort.renamedcontigs.bedgraph.gz | awk '$4==1' | grep -v '^GL\|^NC\|^hs\|^MT' | sed 's/^X/23/;s/^Y/24/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^24/Y/'  | /Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i stdin "$prefix".l100c500.no_ns.refwidenedplus100.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin | awk '$3-$2>10000' > "$prefix"_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed
done

##For diploid assemblies, find regions covered by both haplotypes
/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i HG2_MCha_h0pol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed HG2_MCha_h1pol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed
awk '$4==2' HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '{sum+=$3-$2} END {print sum}' HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '$3-$2>10000' /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed -b stdin > HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed
awk '$3-$2>10000' /Users/jzook/Downloads/segdups_selfchain_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed -b stdin > HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed

/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i HG2_ORod_raw.01_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed HG2_ORod_raw.02_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed
awk '$4==2' HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '$3-$2>10000' /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed -b stdin > HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed
awk '$3-$2>10000' /Users/jzook/Downloads/segdups_selfchain_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.bed -b stdin > HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed


/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i HG2_IFid_pseu2.1_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed HG2_IFid_pseu2.2_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed
awk '$4==2' HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '{sum+=$3-$2} END {print sum}' HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '$3-$2>100' /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed -b stdin > HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed
awk '{sum+=$3-$2} END {print sum}' HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed
awk '$3-$2>10000' /Users/jzook/Downloads/segdups_selfchain_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed -b stdin > HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed

/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i HG2_ACar_cns.h_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed HG2_ACar_cns.p_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed
awk '$4==2' HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.bed > HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed
awk '$3-$2>100' /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.bed -b stdin > HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed
awk '$3-$2>10000' /Users/jzook/Downloads/segdups_selfchain_merged50.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.bed -b stdin > HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed
awk '{sum+=$3-$2} END {print sum}' HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed

/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i HG2_MCha_hbothpol_0218_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed HG2_ORod_raw.both_0518_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt10k.subsegdupgt10k.bed HG2_IFid_pseu2.both_0118_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed HG2_ACar_cns.both_0418_l100c500.qdelta.sort.renamedcontigs.cov1orinvcf.covboth.subrepgt100.subsegdupgt10k.bed > diploidcovmultiintersect.bed
/Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i diploidcovmultiintersect.bed | awk '$3-$2>10000' | awk '{FS=OFS="\t"} { print $1, $2+50, $3-50}' > diploidcov_v0.6.bed
awk '{sum+=$3-$2} END {print sum}' diploidcov_v0.6.bed
awk '$4>1' diploidcovmultiintersect.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin | awk '$3-$2>10000' | awk '{FS=OFS="\t"} { print $1, $2+50, $3-50}' > diploidcov2_v0.6.bed
awk '{sum+=$3-$2} END {print sum}' diploidcov2_v0.6.bed

#expand Tier2 regions to ensure they encompass repeats fully
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50_slop5.bed -b ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset.sort.bed > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_repeatsbedoverlap.bed
/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset.sort.bed ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_repeatsbedoverlap.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin | awk '{FS=OFS="\t"} { print $1, $2-100, $3+100}' > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats.bed

/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -v -a ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats.bed -b ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap.bed
wc -l ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap.bed

#Find Tier2 regions that overlap homref bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap.bed -b diploidcov_v0.6.bed > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap.bed
wc -l  ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap.bed


/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -header -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b diploidcov_v0.6.bed | awk '$7=="PASS"'  | wc -l
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -header -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b diploidcov2_v0.6.bed | awk '$7=="PASS"'  | wc -l


#re
grep HG2 ../../triounion_171212/truvaritest06asm222PASSs30p0/fp.vcf | grep -v 'NumClusterSVs=1;\|NumClusterSVs=2;' | grep 'ACar\|ORod\|MCha\|PB\|MHAP\|Falc' | sed 's/.*REFWIDENED=\(.*\):\(.*\)-\(.*\);ALTWIDENED.*/\1	\2	\3/' | awk '{FS=OFS="\t"} { print $1, $2-1000, $3+1000}' | /Applications/bioinfo/bedtools2.26.0/bin/intersectBed -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b stdin | awk '$7=="PASS"'  | wc -l
grep HG2 ../../triounion_171212/truvaritest06asm222PASSs30p0/fp.vcf | grep -v 'NumClusterSVs=1;\|NumClusterSVs=2;' | grep 'ACar\|ORod\|MCha\|PB\|MHAP\|Falc' | sed 's/.*REFWIDENED=\(.*\):\(.*\)-\(.*\);ALTWIDENED.*/\1	\2	\3/' | awk '{FS=OFS="\t"} { print $1, $2-1000, $3+1000}' | sed 's/^X/23/;s/^Y/24/;s/^MT/25/' | sort -k1,1n -k2,2n -k3,3n | sed 's/^23/X/;s/^24/Y/;s/^25/MT/' | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin   > ../../triounion_171212/truvaritest06asm222PASSs30p0/fp.HG2.clustergt2.longreadsupport.slop1000.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap.bed -b ../../triounion_171212/truvaritest06asm222PASSs30p0/fp.HG2.clustergt2.longreadsupport.slop1000.bed > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap_asmunmatched.bed

#zgrep -v ^# ../../NHGRI_SVrefine_04122018/HG2_unmatched.2.2.2.clustered.longreadsupport.vcf.gz | grep -v 'NumClusterSVs=1;' | sed 's/.*REFWIDENED=\(.*\):\(.*\)-\(.*\);ALTPOS.*/\1	\2	\3/' | awk '{FS=OFS="\t"} { print $1, $2-1000, $3+1000}' | /Applications/bioinfo/bedtools2.26.0/bin/intersectBed -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b stdin | awk '$7=="PASS"'  | wc -l
#zgrep -v ^# ../../NHGRI_SVrefine_04122018/HG2_unmatched.2.2.2.clustered.longreadsupport.vcf.gz | grep -v 'NumClusterSVs=1;' | sed 's/.*REFWIDENED=\(.*\):\(.*\)-\(.*\);ALTPOS.*/\1	\2	\3/' | awk '{FS=OFS="\t"} { print $1, $2-1000, $3+1000}'  | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin   > ../../NHGRI_SVrefine_04122018/HG2_unmatched.2.2.2.clustered.longreadsupport.slop1000.bed
#/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap.bed -b ../../NHGRI_SVrefine_04122018/HG2_unmatched.2.2.2.clustered.longreadsupport.slop1000.bed > ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap_homrefbedoverlap_asmunmatched.bed

#subtract tier2 not in tier1 and unmatched assembly calls from diploidcov.bed
/Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a diploidcov_v0.6.bed -b ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a stdin -b ../../triounion_171212/truvaritest06asm222PASSs30p0/fp.HG2.clustergt2.longreadsupport.slop1000.bed >  diploidcov_noTier2_noasmunmatched_v0.6.bed
awk '{sum+=$3-$2} END {print sum}' diploidcov_noTier2_noasmunmatched_v0.6.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -header -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b diploidcov_noTier2_noasmunmatched_v0.6.bed | awk '$7=="PASS"'  | wc -l

zgrep -v ^# ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '{FS=OFS="\t"} { if($7=="ClusteredCalls" || $7=="lt50bp" || $7=="MaxEditDistgt04") print $1, $2-50, $2+length($4)+50}' | sed 's/^X/23/;s/^Y/24/;s/^MT/25/' | sort -k1,1n -k2,2n -k3,3n | sed 's/^23/X/;s/^24/Y/;s/^25/MT/' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered.bed
zgrep -v ^# ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '{FS=OFS="\t"} { if($7=="NoConsensusGT") print $1, $2-50, $2+length($4)+50}' | sed 's/^X/23/;s/^Y/24/;s/^MT/25/' | sort -k1,1n -k2,2n -k3,3n | sed 's/^23/X/;s/^24/Y/;s/^25/MT/' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50_slop5.bed -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered.bed | /Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i stdin ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 | awk '{FS=OFS="\t"} { print $1, $2-50, $3+50}' | awk '{FS=OFS="\t"} { if($2<0) $2=0; print}' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered_repeatexpanded.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50_slop5.bed -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT.bed | /Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i stdin ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 | awk '{FS=OFS="\t"} { print $1, $2-50, $3+50}' | awk '{FS=OFS="\t"} { if($2<0) $2=0; print}' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_repeatexpanded.bed
gunzip -c ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '($1 ~ /^#/) || $7=="PASS"' | /Applications/bioinfo/bedtools2.26.0/bin/intersectBed -v -a ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_repeatexpanded.bed -b stdin > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_repeatexpanded_noPASSoverlap.bed
gunzip -c ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '($1 ~ /^#/) || $7=="PASS"' | /Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a stdin -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered_repeatexpanded.bed | wc -l

/Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a diploidcov_noTier2_noasmunmatched_v0.6.bed -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_repeatexpanded_noPASSoverlap.bed | /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a stdin -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered_repeatexpanded.bed >  diploidcov_noTier2_noasmunmatched_nolt50bp_noclustered_nonoconsGT_v0.6.bed
awk '{sum+=$3-$2} END {print sum}' diploidcov_noTier2_noasmunmatched_nolt50bp_noclustered_nonoconsGT_v0.6.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -header -a ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz -b diploidcov_noTier2_noasmunmatched_nolt50bp_noclustered_nonoconsGT_v0.6.bed | awk '$7=="PASS"'  | wc -l

#number of PASS calls not in bed due to nearby variant <50bp
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -a /Users/jzook/Documents/AJTrio/SVs/triounion_171212/NIST_SVs_Integration_v0.6/HG002_SVs_Tier1_v0.6.vcf.gz -b /Users/jzook/Documents/AJTrio/SVs/triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered.bed | awk '$7=="PASS"' | wc -l
#    1837
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -a /Users/jzook/Documents/AJTrio/SVs/triounion_171212/NIST_SVs_Integration_v0.6/HG002_SVs_Tier1_v0.6.vcf.gz -b /Users/jzook/Documents/AJTrio/SVs/triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT.bed -header |  /Applications/bioinfo/bedtools2.26.0/bin/subtractBed -a stdin -b /Users/jzook/Documents/AJTrio/SVs/triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_lt50orClustered.bed | awk '$7=="PASS"' | wc -l
# 856

#Form Tier 2 bed
zgrep -v ^# ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '{FS=OFS="\t"} { if($7=="ClusteredCalls" || $7=="MaxEditDistgt04") print $1, $2-1000, $2+length($4)+1000}'  | sed 's/^X/23/;s/^Y/24/;s/^MT/25/' | sort -k1,1n -k2,2n -k3,3n | sed 's/^23/X/;s/^24/Y/;s/^25/MT/'  | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_mexedit40orClustered.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50_slop5.bed -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_mexedit40orClustered.bed | /Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i stdin ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_mexedit40orClustered.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 | awk '{FS=OFS="\t"} { print $1, $2-50, $3+50}' | awk '{FS=OFS="\t"} { if($2<0) $2=0; print}' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_mexedit40orClustered_repeatexpanded.bed
zgrep -v ^# ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '{FS=OFS="\t"} { if($7=="NoConsensusGT" && (length($4)>49 || length($5)>49)) print $1, $2-50, $2+length($4)+50}' | sed 's/^X/23/;s/^Y/24/;s/^MT/25/' | sort -k1,1n -k2,2n -k3,3n | sed 's/^23/X/;s/^24/Y/;s/^25/MT/' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp.bed
/Applications/bioinfo/bedtools2.26.0/bin/intersectBed -wa -a /Users/jzook/Downloads/repeats_trfSimplerepLowcomplex_merged50_slop5.bed -b ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp.bed | /Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i stdin ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 | awk '{FS=OFS="\t"} { print $1, $2-50, $3+50}' | awk '{FS=OFS="\t"} { if($2<0) $2=0; print}' > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp_repeatexpanded.bed
gunzip -c ../../triounion_171212/svanalyzer_union_171212_Tier1_HG2_v0.6.vcf.gz | awk '($1 ~ /^#/) || $7=="PASS"' | /Applications/bioinfo/bedtools2.26.0/bin/intersectBed -v -a ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp_repeatexpanded.bed -b stdin > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp_repeatexpanded_noPASSoverlap.bed
/Applications/bioinfo/bedtools2.26.0/bin/multiIntersectBed -i ../../triounion_171212/union_171212_Tier2_unknownsequence_HG002_2techor4callset_plusrepeats_noTier1overlap.bed ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_noConsGT_gt49bp_repeatexpanded_noPASSoverlap.bed ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier1PASS_HG2_mexedit40orClustered_repeatexpanded.bed | /Applications/bioinfo/bedtools2.26.0/bin/mergeBed -i stdin -d 50 > ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier2_HG2.bed
wc -l ../../triounion_171212/svanalyzer_union_171212_v0.6_Tier2_HG2.bed
