#HG001 10X mapped with GRCh38
#GATK Haplotype caller
#Run 9/22/16  Ran ~16-56 min
#chrX run on 10/5

# HP1
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_1.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_1.bai -ioutput_prefix=HG001_1_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_2.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_2.bai -ioutput_prefix=HG001_2_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_3.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_3.bai -ioutput_prefix=HG001_3_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_4.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_4.bai -ioutput_prefix=HG001_4_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_5.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_5.bai -ioutput_prefix=HG001_5_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_6.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_6.bai -ioutput_prefix=HG001_6_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_7.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_7.bai -ioutput_prefix=HG001_7_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_8.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_8.bai -ioutput_prefix=HG001_8_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_9.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_9.bai -ioutput_prefix=HG001_9_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_10.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_10.bai -ioutput_prefix=HG001_10_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_11.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_11.bai -ioutput_prefix=HG001_11_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_12.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_12.bai -ioutput_prefix=HG001_12_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_13.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_13.bai -ioutput_prefix=HG001_13_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_14.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_14.bai -ioutput_prefix=HG001_14_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_15.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_15.bai -ioutput_prefix=HG001_15_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_16.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_16.bai -ioutput_prefix=HG001_16_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_17.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_17.bai -ioutput_prefix=HG001_17_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_18.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_18.bai -ioutput_prefix=HG001_18_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_19.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_19.bai -ioutput_prefix=HG001_19_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_20.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_20.bai -ioutput_prefix=HG001_20_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_21.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_21.bai -ioutput_prefix=HG001_21_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_22.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_22.bai -ioutput_prefix=HG001_22_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_X.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP1_X.bai -ioutput_prefix=HG001_X_10X_HP1_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chrX -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/

# HP2
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_1.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_1.bai -ioutput_prefix=HG001_1_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_2.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_2.bai -ioutput_prefix=HG001_2_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_3.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_3.bai -ioutput_prefix=HG001_3_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_4.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_4.bai -ioutput_prefix=HG001_4_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_5.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_5.bai -ioutput_prefix=HG001_5_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_6.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_6.bai -ioutput_prefix=HG001_6_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_7.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_7.bai -ioutput_prefix=HG001_7_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_8.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_8.bai -ioutput_prefix=HG001_8_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_9.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_9.bai -ioutput_prefix=HG001_9_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_10.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_10.bai -ioutput_prefix=HG001_10_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_11.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_11.bai -ioutput_prefix=HG001_11_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_12.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_12.bai -ioutput_prefix=HG001_12_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_13.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_13.bai -ioutput_prefix=HG001_13_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_14.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_14.bai -ioutput_prefix=HG001_14_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_15.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_15.bai -ioutput_prefix=HG001_15_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_16.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_16.bai -ioutput_prefix=HG001_16_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_17.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_17.bai -ioutput_prefix=HG001_17_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_18.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_18.bai -ioutput_prefix=HG001_18_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_19.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_19.bai -ioutput_prefix=HG001_19_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_20.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_20.bai -ioutput_prefix=HG001_20_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_21.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_21.bai -ioutput_prefix=HG001_21_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_22.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_22.bai -ioutput_prefix=HG001_22_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chr22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35-anyref -isorted_bam=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_X.bam -isorted_bai=/HG001/GRCh38/10XGenomics/HG001_GRCh38_10X_HP2_X.bai -ioutput_prefix=HG001_X_10X_HP2_GATKHC_gvcf -iref=/assets/GRCh38hs38d1noalt.fasta-index.tar.gz -iextra_options="-L chrX -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator" --destination=/HG001/GRCh38/10XGenomics/GATKHC_output/
