
#GATKHC for 10X HG004
#HP1 and HP2 Run 8/22/16, ran 2-38 min

# Haplotype 1 = HP1
# No MT bam so MT not run 
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.1.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.1.bai -ioutput_prefix=HG004_1_10X_HP1_GATKHC_gvcf -iextra_options="-L 1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.2.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.2.bai -ioutput_prefix=HG004_2_10X_HP1_GATKHC_gvcf -iextra_options="-L 2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.3.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.3.bai -ioutput_prefix=HG004_3_10X_HP1_GATKHC_gvcf -iextra_options="-L 3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.4.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.4.bai -ioutput_prefix=HG004_4_10X_HP1_GATKHC_gvcf -iextra_options="-L 4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.5.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.5.bai -ioutput_prefix=HG004_5_10X_HP1_GATKHC_gvcf -iextra_options="-L 5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.6.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.6.bai -ioutput_prefix=HG004_6_10X_HP1_GATKHC_gvcf -iextra_options="-L 6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.7.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.7.bai -ioutput_prefix=HG004_7_10X_HP1_GATKHC_gvcf -iextra_options="-L 7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.8.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.8.bai -ioutput_prefix=HG004_8_10X_HP1_GATKHC_gvcf -iextra_options="-L 8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.9.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.9.bai -ioutput_prefix=HG004_9_10X_HP1_GATKHC_gvcf -iextra_options="-L 9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.10.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.10.bai -ioutput_prefix=HG004_10_10X_HP1_GATKHC_gvcf -iextra_options="-L 10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.11.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.11.bai -ioutput_prefix=HG004_11_10X_HP1_GATKHC_gvcf -iextra_options="-L 11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.12.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.12.bai -ioutput_prefix=HG004_12_10X_HP1_GATKHC_gvcf -iextra_options="-L 12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.13.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.13.bai -ioutput_prefix=HG004_13_10X_HP1_GATKHC_gvcf -iextra_options="-L 13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.14.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.14.bai -ioutput_prefix=HG004_14_10X_HP1_GATKHC_gvcf -iextra_options="-L 14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.15.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.15.bai -ioutput_prefix=HG004_15_10X_HP1_GATKHC_gvcf -iextra_options="-L 15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.16.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.16.bai -ioutput_prefix=HG004_16_10X_HP1_GATKHC_gvcf -iextra_options="-L 16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.17.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.17.bai -ioutput_prefix=HG004_17_10X_HP1_GATKHC_gvcf -iextra_options="-L 17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.18.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.18.bai -ioutput_prefix=HG004_18_10X_HP1_GATKHC_gvcf -iextra_options="-L 18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.19.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.19.bai -ioutput_prefix=HG004_19_10X_HP1_GATKHC_gvcf -iextra_options="-L 19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.20.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.20.bai -ioutput_prefix=HG004_20_10X_HP1_GATKHC_gvcf -iextra_options="-L 20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.21.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.21.bai -ioutput_prefix=HG004_21_10X_HP1_GATKHC_gvcf -iextra_options="-L 21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.22.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.22.bai -ioutput_prefix=HG004_22_10X_HP1_GATKHC_gvcf -iextra_options="-L 22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.X.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.X.bai -ioutput_prefix=HG004_X_10X_HP1_GATKHC_gvcf -iextra_options="-L X -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.Y.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP1.Y.bai -ioutput_prefix=HG004_Y_10X_HP1_GATKHC_gvcf -iextra_options="-L Y -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/

# Haplotype 2 = HP2
# No MT bam so MT not run
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.1.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.1.bai -ioutput_prefix=HG004_1_10X_HP2_GATKHC_gvcf -iextra_options="-L 1 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.2.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.2.bai -ioutput_prefix=HG004_2_10X_HP2_GATKHC_gvcf -iextra_options="-L 2 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.3.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.3.bai -ioutput_prefix=HG004_3_10X_HP2_GATKHC_gvcf -iextra_options="-L 3 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.4.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.4.bai -ioutput_prefix=HG004_4_10X_HP2_GATKHC_gvcf -iextra_options="-L 4 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.5.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.5.bai -ioutput_prefix=HG004_5_10X_HP2_GATKHC_gvcf -iextra_options="-L 5 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.6.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.6.bai -ioutput_prefix=HG004_6_10X_HP2_GATKHC_gvcf -iextra_options="-L 6 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.7.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.7.bai -ioutput_prefix=HG004_7_10X_HP2_GATKHC_gvcf -iextra_options="-L 7 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.8.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.8.bai -ioutput_prefix=HG004_8_10X_HP2_GATKHC_gvcf -iextra_options="-L 8 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.9.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.9.bai -ioutput_prefix=HG004_9_10X_HP2_GATKHC_gvcf -iextra_options="-L 9 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.10.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.10.bai -ioutput_prefix=HG004_10_10X_HP2_GATKHC_gvcf -iextra_options="-L 10 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.11.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.11.bai -ioutput_prefix=HG004_11_10X_HP2_GATKHC_gvcf -iextra_options="-L 11 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.12.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.12.bai -ioutput_prefix=HG004_12_10X_HP2_GATKHC_gvcf -iextra_options="-L 12 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.13.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.13.bai -ioutput_prefix=HG004_13_10X_HP2_GATKHC_gvcf -iextra_options="-L 13 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.14.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.14.bai -ioutput_prefix=HG004_14_10X_HP2_GATKHC_gvcf -iextra_options="-L 14 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.15.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.15.bai -ioutput_prefix=HG004_15_10X_HP2_GATKHC_gvcf -iextra_options="-L 15 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.16.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.16.bai -ioutput_prefix=HG004_16_10X_HP2_GATKHC_gvcf -iextra_options="-L 16 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.17.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.17.bai -ioutput_prefix=HG004_17_10X_HP2_GATKHC_gvcf -iextra_options="-L 17 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.18.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.18.bai -ioutput_prefix=HG004_18_10X_HP2_GATKHC_gvcf -iextra_options="-L 18 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.19.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.19.bai -ioutput_prefix=HG004_19_10X_HP2_GATKHC_gvcf -iextra_options="-L 19 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.20.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.20.bai -ioutput_prefix=HG004_20_10X_HP2_GATKHC_gvcf -iextra_options="-L 20 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.21.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.21.bai -ioutput_prefix=HG004_21_10X_HP2_GATKHC_gvcf -iextra_options="-L 21 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.22.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.22.bai -ioutput_prefix=HG004_22_10X_HP2_GATKHC_gvcf -iextra_options="-L 22 -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.X.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.X.bai -ioutput_prefix=HG004_X_10X_HP2_GATKHC_gvcf -iextra_options="-L X -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/
dx run -y GIAB:/Workflow/GATK_V3.5/gatk-haplotypecaller-v35 -isorted_bam=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.Y.bam -isorted_bai=GIAB:/HG004/10XGenomics/NA24143_GRCh37_HP2.Y.bai -ioutput_prefix=HG004_Y_10X_HP2_GATKHC_gvcf -iextra_options="-L Y -stand_call_conf 2 -stand_emit_conf 2 -A BaseQualityRankSumTest -A ClippingRankSumTest -A Coverage -A FisherStrand -A LowMQ -A RMSMappingQuality -A ReadPosRankSumTest -A StrandOddsRatio -A HomopolymerRun -A TandemRepeatAnnotator"  --destination=/HG004/10XGenomics/GATKHC_output/

